### PR TITLE
Shift pairings init draft

### LINKED ISSUE #6: Keep track of past pairings
We should keep track of past pairings so that `generatePairings` (in `utils/`) doesn't pair the same people twice.

- If we have to pair people again (unlikely) send a cheeky message saying "oops! sorry! deal with it!"
- Write basic JSON or text file or something with pairings (must be sophisticated), so that the "checking up" functionality can do its job

### PR DESCRIPTION
**Pull request by zefwang**
Closes #6 in a roundabout way lmao

### CODE REVIEW

File: utils/generatePairs.ts [hunk 1]
```diff
@@ -50,4 +50,31 @@ function shuffle(array: string[]): string[] {
...
+  const firstPersonList = lastPairings.map((pairs: string[]) => pairs[0])
+
+  // put first person at end of list (shifting everyone up one)
+  const firstElem = firstPersonList.shift()
+  if (firstElem) {
+    firstPersonList.push(firstElem)
+  }
+
+  // update pairings with new first person
+  lastPairings = lastPairings.map((pairs: string[], ind: number) => [firstPersonList[ind], pairs[1]])
```
Comment (VKong6019) [inline utils/generatePairs.ts:67]: does this work for odd number of people? specifically, in the case where a pairing could have 3 members

File: __tests__/generatePairs.test.ts [hunk 2]
```diff
@@ -76,4 +76,16 @@ describe('testing the util function for generating pairs', () =>{
         expect(actualList).toStrictEqual([]);
         expect(mockConversations.members).toHaveBeenCalled();
     })
-})
\ No newline at end of file
+
+    it('should shift member by one', async () => {
+        jest.mock('../utils/data/lastPairings.json', ()=>([
```
Comment (VKong6019) [inline __tests__/generatePairs.test.ts:81]: chungus

File: utils/generatePairs.ts [hunk 3]
```diff
@@ -50,4 +50,31 @@ function shuffle(array: string[]): string[] {
   return array;
 }
 
-export default generatePairs;
\ No newline at end of file
+export const shiftByOne = async (): Promise<string[][]>  => {
```
Comment (VKong6019) [inline utils/generatePairs.ts:53]: would `shiftByOne` be used inside the `generatePairs` function above?

File: utils/generatePairs.ts (hunk 1 above)
Comment (zefwang) [reply utils/generatePairs.ts:67]: O lmao that's so funny the 2nd and 3rd person will always be together :)

Comment (zefwang) [reply utils/generatePairs.ts:67]: I propose we just don't let there be an odd number of people

File: utils/generatePairs.ts (hunk 3 above)
Comment (zefwang) [reply utils/generatePairs.ts:53]: I think it would just be called instead after the first call no?

Comment (VKong6019) [reply utils/generatePairs.ts:53]: ohi see

File: utils/generatePairs.ts [hunk 4]
```diff
@@ -55,17 +55,29 @@ export const shiftByOne = async (): Promise<string[][]>  => {
...
+  firstPersonList.push(firstElem)
 
   // update pairings with new first person
   lastPairings = lastPairings.map((pairs: string[], ind: number) => [firstPersonList[ind], pairs[1]])
 
+  // get last person
+  let oddPersonOut: string;
+  if (oddGroup) {
+    oddPersonOut = oddGroup[2];
+    const oddPersonIndex = Math.floor(Math.random() * lastPairings.length);
```
Comment (VKong6019) [inline utils/generatePairs.ts:74]: rng FTW

File: __tests__/generatePairs.test.ts [hunk 5]
```diff
@@ -8,6 +8,13 @@ describe('testing the util function for generating pairs', () =>{
             }
         }
     };
+    beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (zefwang) [reply __tests__/generatePairs.test.ts:11]: 4530 TA levels of Jest expertise lfg

**Review by VKong6019 — APPROVED** [review]
Amazing

**Pull request merged**